sympy/plotting/plot.py

```diff
@@ -2135,7 +2135,22 @@ def check_arguments(args, expr_len, nb_of_free_symbols):
             i = len(args) + 1
 
         exprs = Tuple(*args[:i])
-        free_symbols = list(set().union(*[e.free_symbols for e in exprs]))
+        free_symbols = list(set().union(*[e.free_symbols for e in exprs])) # sum(args[:expr_len]).free_symbols
+
+        #Warn: Dimension
+        variables = set(free_symbols) # get all variables
+        specify_symbols = set([s[0] for s in args[expr_len:]]) # get symbols from all ranges
+        all_symbols = variables | specify_symbols
+        if len(all_symbols) > len(variables):
+            msj = "Dimension Error: there be a variable ranged that ins´t part of any ecuation!"
+            raise Exception(msj)
+        if len(all_symbols) > nb_of_free_symbols:
+            if specify_symbols == set():
+                msj =  "\nYou should be use ONLY %s (%s) parameters from %s"%("one" if nb_of_free_symbols is 1 else "two" if nb_of_free_symbols is 2 else "", nb_of_free_symbols, all_symbols)
```

Comment:
The newline at the beginning of the message is not necessary. It's also bad form to use the second person ("you") in error messages. Keep in mind that the person seeing the error message may not be the person who wrote the code that caused it.